### ISSUE TITLE
Sigma on single-page apps: plugin destroyed automatically when sigma is killed?

### ISSUE THREAD

**Issue by sheymann** [jacomyal/sigma.js#379]
I'm implementing Sigma + plugins in an Angular.js environment, where the user can go and leave the "page" where the Sigma instance draws the graph. It means that the application instanciates Sigma each time the user goes to the "page", and kills Sigma each time the user leaves the "page". It's fine until I use plugins: those who are instanciated with a Sigma instance need a proper kill method. I can manage it with my application code and by forking some official plugins but I feel that Sigma could provide helpers in managing the plugins lifecycle. Singe-page apps become a standard I'm not probably the only one with this need.

So basically it would be great if we could **register** plugins in Sigma instances so that they are killed automatically when a Sigma instance is killed.
